internal/bson/document.go

```diff
@@ -247,10 +298,7 @@ func (doc *Document) ReadFrom(r *bufio.Reader) error {
 		}
 	}
 
-	if _, err := types.ConvertDocument(doc); err != nil {
-		return lazyerrors.Errorf("bson.Document.ReadFrom: %w", err)
-	}
-
+	*doc = Document{fields: fields}
```

Comment:
That's fine for now, but note that previous code reused doc.m and doc.keys' capacities, and this new code replaces the whole document